src/lib.rs

```diff
@@ -134,15 +139,13 @@ impl<T: Send> State<T> {
             let seq = unsafe { (*node.get()).sequence.load(Acquire) };
             let diff: isize = seq as isize - (pos + 1) as isize;
             if diff == 0 {
```

Comment:
Probably pre-existing, but clippy is suggesting we change this:
```
warning: `if` chain can be rewritten with `match`
   --> src/lib.rs:141:13
    |
141 | /             if diff == 0 {
142 | |                 match self.dequeue_pos.compare_exchange_weak(pos, pos + 1, Relaxed, Relaxed) {
143 | |                     Ok(_old_pos) => unsafe {
144 | |                         let value = (*node.get()).value.take();
...   |
153 | |                 pos = self.dequeue_pos.load(Relaxed);
154 | |             }
    | |_____________^
    |
    = help: consider rewriting the `if` chain to use `cmp` and `match`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#comparison_chain
```

Reply:
Seems to be pre-existing, but yeah I can address it now. Might as well. 

I followed the clippy link, but am not quite sure what the recommended change is. Is it suggesting we use a match statement for the conditionals checking the value of `diff`? 
e.g., this:
```rust
match diff {
    0 => match self.enqueue_pos.compare... { }
    x if x < 0 => return Err(value),
    _ => pos = self.enqueue_pos.load(Relaxed),
}
```

Or is it suggesting some other kind of odd combination of the two match statements shown above?